apps/confidential/confidential_test.go

```diff
@@ -246,6 +246,76 @@ func TestAcquireTokenByAuthCode(t *testing.T) {
 		t.Fatalf("unexpected access token %s", tk.AccessToken)
 	}
 	account := client.Account(tk.Account.HomeAccountID)
+	if account.HomeAccountID != "123-456.fake" {
+		t.Fatal("Unexpected Account.HomeAccountId")
+	}
+	if account.PreferredUsername != "fakeuser@fakeplace.fake" {
+		t.Fatal("Unexpected Account.PreferredUsername")
+	}
+	// second attempt should return the cached token
+	tk, err = client.AcquireTokenSilent(context.Background(), tokenScope, WithSilentAccount(account))
+	if err != nil {
+		t.Fatal(err)
+	}
+	if tk.AccessToken != token {
+		t.Fatalf("unexpected access token %s", tk.AccessToken)
+	}
+}
+
+func TestAcquireTokenByAuthCodeADFS(t *testing.T) {
+	cred, err := NewCredFromSecret("fake_secret")
+	if err != nil {
+		t.Fatal(err)
+	}
+	client, err := fakeClient(accesstokens.TokenResponse{
```

Comment:
What quirks of the ADFS auth code flow should this test cover? This fake replaces the real code for authority validation, authentication, caching, etc., so I wonder whether this test would catch a production bug related to those quirks.

Reply:
This test just checks if homeAccountId is being properly set to oid.oid or oid.uid, as suggested here 
>HomeAccountId property of Account objects from should be in format "oid.oid" (for AAD it's "oid.tid" but we don't have "tid" in ADFS). This is pretty important, otherwise other MSALs might not be able to read from the cache.